fullnode/fullnode_lazy.go

```diff
@@ -0,0 +1,235 @@
+package fullnode
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	cdctypes "github.com/cosmos/cosmos-sdk/codec/types"
+	cryptocodec "github.com/cosmos/cosmos-sdk/crypto/codec"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	"github.com/cosmos/cosmos-sdk/types"
+	accounttypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	apptypes "github.com/pokt-network/poktroll/x/application/types"
+	servicetypes "github.com/pokt-network/poktroll/x/service/types"
+	sessiontypes "github.com/pokt-network/poktroll/x/session/types"
+
+	sdk "github.com/pokt-network/shannon-sdk"
+)
+
+// queryCodec is a package-level codec for unmarshaling account data.
+// Initialized globally to avoid expensive repeated setup across LazyFullNode instances.
+var queryCodec *codec.ProtoCodec
```

Comment:
I feel like this belongs in the `types` package: https://github.com/pokt-network/shannon-sdk/tree/main/types

Reply:
I've moved it there and imported it into the full node